crates/mdk-core/src/image_processing/metadata.rs

```diff
@@ -0,0 +1,384 @@
+                })?;
+        }
+        "image/png" => {
+            let encoder = PngEncoder::new(&mut output);
+            encoder
+                .write_image(
+                    img.as_bytes(),
+                    img.width(),
+                    img.height(),
+                    img.color().into(),
+                )
+                .map_err(|e| ImageProcessingError::MetadataExtractionFailed {
+                    reason: format!("Failed to re-encode PNG: {}", e),
+                })?;
+        }
+        _ => {
+            // For unknown formats, return error
+            return Err(ImageProcessingError::MetadataExtractionFailed {
+                reason: format!("Unsupported image format for EXIF stripping: {}", mime_type),
+            });
+        }
+    }
+
+    Ok((output.into_inner(), img))
+}
+
+/// Apply EXIF orientation transform to an image
+///
+/// Reads the EXIF orientation tag from the original image data and applies
+/// the appropriate rotation and/or flip operations to the decoded image.
+/// This ensures images display correctly even after EXIF metadata is stripped.
+///
+/// EXIF Orientation values:
+/// 1 = Normal
+/// 2 = Flip horizontal
+/// 3 = Rotate 180°
+/// 4 = Flip vertical
+/// 5 = Flip horizontal + Rotate 270° CW
+/// 6 = Rotate 90° CW
+/// 7 = Flip horizontal + Rotate 90° CW
+/// 8 = Rotate 270° CW
+fn apply_exif_orientation(
+    data: &[u8],
+    img: image::DynamicImage,
+) -> Result<image::DynamicImage, ImageProcessingError> {
+    // Try to read EXIF data - if it fails or doesn't exist, just return the original image
+    let exif_reader = match Reader::new().read_from_container(&mut Cursor::new(data)) {
+        Ok(exif) => exif,
+        Err(_) => return Ok(img), // No EXIF data or couldn't read it - return as-is
+    };
+
+    // Get the orientation tag
+    let orientation = match exif_reader.get_field(Tag::Orientation, In::PRIMARY) {
+        Some(field) => match field.value.get_uint(0) {
+            Some(val) => val,
+            None => return Ok(img), // Couldn't parse orientation - return as-is
+        },
+        None => return Ok(img), // No orientation tag - return as-is
+    };
+
+    // Apply the appropriate transform based on orientation value
+    let transformed = match orientation {
+        1 => img,                     // Normal - no transformation needed
+        2 => img.fliph(),             // Flip horizontal
+        3 => img.rotate180(),         // Rotate 180°
+        4 => img.flipv(),             // Flip vertical
+        5 => img.rotate270().fliph(), // Flip horizontal + Rotate 270° CW
+        6 => img.rotate90(),          // Rotate 90° CW
+        7 => img.rotate90().fliph(),  // Flip horizontal + Rotate 90° CW
+        8 => img.rotate270(),         // Rotate 270° CW (or 90° CCW)
+        _ => img,                     // Unknown orientation value - return as-is
+    };
+
+    Ok(transformed)
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
We should add tests for all public methods, I think we're missing: `strip_exif_and_return_image`, `is_safe_raster_format` and `preflight_dimension_check`